tool/src/js/formatter.rs

```diff
@@ -43,6 +43,66 @@ const RESERVED: &[&str] = &[
     "with",
 ];
 
+/// From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects
```

Comment:
Issue: most of these names are _valid in JS_ to have types named the same thing. There are a few where it is not actually valid, such as `NaN`.

```javascript
let NaN = "abc"; // this doesn't run
let Date = "abc"; // this is discouraged but fine
```

When the types are in a module, it is common and expected for types named things like `Date` and `Error` to exist. It's wrong to forbid them.